app/modules/power/scripts/power

```diff
@@ -0,0 +1,376 @@
+#! /usr/bin/python
+# power script by tuxudo for MunkiReport
+
+import subprocess
+import plistlib
+import os
+import sys
+import string
+import re
+
+def get_battery_profiler():
+    '''Uses system profiler to get power info for this machine.'''
+    cmd = ['/usr/sbin/system_profiler', 'SPPowerDataType', '-xml']
+    proc = subprocess.Popen(cmd, shell=False, bufsize=-1,
+                            stdin=subprocess.PIPE,
+                            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    (output, unused_error) = proc.communicate()
+    plist = plistlib.readPlistFromString(output)
+
+    # system_profiler xml is an array
+    sp_dict = plist[0]
+    items = sp_dict['_items']
+
+    batteryprofiler = {}
+
+    for obj in items:
+        for item in obj:
+            if item == 'sppower_current_amperage':
+                batteryprofiler['AmperagemA'] = format((float(obj[item])/1000),'.3f')
+            elif item == 'sppower_current_voltage':
+                batteryprofiler['Voltage'] = format((float(obj[item])/1000),'.3f')
+            elif item == 'sppower_battery_charger_connected':
+                batteryprofiler['ExternalConnected'] = obj[item]
+            elif item == 'sppower_ac_charger_ID':
+                batteryprofiler['AdapterID'] = obj[item]
+            elif item == 'sppower_ac_charger_watts':
+                batteryprofiler['attage'] = obj[item] # This is not a typo
+            elif item == 'sppower_ac_charger_family':
+                batteryprofiler['FamilyCode'] = obj[item]
+            elif item == 'sppower_ac_charger_serial_number':
+                batteryprofiler['SerialNumber'] = obj[item]
+            elif item == 'sppower_ups_installed':
+                batteryprofiler['UPSInstalled'] = obj[item]
+            elif item == 'sppower_battery_installed' and obj['sppower_battery_installed'] == "TRUE":
+                batteryprofiler['condition'] = obj['sppower_battery_health_info']['sppower_battery_health']
+            elif item == 'sppower_battery_installed' and obj['sppower_battery_installed'] == "FALSE":
+                batteryprofiler['condition'] = "No Battery"
+            elif item == 'sppower_battery_charge_info':
+                try:
+                    batteryprofiler['MaxCapacity'] = obj[item]['sppower_battery_max_capacity']
```

Comment:
If you are trying to get a property of a dictionary and you're not sure that it exists, use get(). So in this case: 

```python
batteryprofiler['MaxCapacity'] = obj[item].get('sppower_battery_max_capacity', null)
```

https://www.programiz.com/python-programming/methods/dictionary/get

Reply:
That does not have the desired result of not having the key at all in the resulting plist.

Reply:
Why do you set the error to an empty string?

Reply:
That line is there just to give Python something to do. The error isn't used in the model